src/Sylius/Bundle/ApiBundle/Doctrine/QueryItemExtension/OrderMethodsItemExtension.php

```diff
@@ -79,7 +79,14 @@ private function applyForVisitor(QueryBuilder $queryBuilder, QueryNameGeneratorI
         $queryBuilder
             ->leftJoin(sprintf('%s.customer', $rootAlias), 'customer')
             ->leftJoin('customer.user', 'user')
-            ->andWhere($queryBuilder->expr()->orX('user IS NULL', sprintf('%s.customer IS NULL', $rootAlias)))
+            ->andWhere($queryBuilder->expr()->orX(
+                'user IS NULL',
+                sprintf('%s.customer IS NULL', $rootAlias),
+                $queryBuilder->expr()->andX(
+                    sprintf('%s.customer IS NOT NULL', $rootAlias),
+                    sprintf('%s.createdByGuest = :createdByGuest', $rootAlias)
+                )
+            ))->setParameter('createdByGuest', true)
```

Comment:
This is the same as in the previous repository, maybe we should introduce some custom doctrine filter/function for that? Again, an improvement to be considered 🖖

Reply:
I would like to open improvement PR with doctrine criteria for this later